src/GovUk.Education.ExploreEducationStatistics.Common/Utils/Html/HtmlToTextConverter.cs

```diff
@@ -36,15 +36,18 @@ private string Output()
         {
             var rawLines = _builder.ToString().ToLines();
 
-            // Run post-filtering on lines to remove any extraneous
-            // whitespace from the end of lines. This is mostly for
-            // better compat with editors/IDEs, where whitespace may
-            // be automatically removed upon file save.
-            var filteredBuilder = new StringBuilder();
-
-            rawLines.ForEach(line => filteredBuilder.AppendLine(line.TrimEnd()));
-
-            return filteredBuilder.ToString().TrimEnd();
+            // Run post-filtering on lines to:
+            // 1. Remove any extraneous whitespace from the end of lines. This is
+            // mostly for better compatibility with formatters / editors / IDEs, where
+            // trailing whitespace may be automatically removed upon file save.
+            // 2. Ensure we have CRLF (\r\n) line endings for better compatibility with
+            // our Windows users (the majority of users). Otherwise, line endings
+            // would be server OS dependent and this can lead to inconsistent files.
+            var postFilterBuilder = new StringBuilder();
+
+            rawLines.ForEach(line => postFilterBuilder.AppendCrlfLine(line.TrimEnd()));
+
+            return postFilterBuilder.ToString().TrimEnd();
```

Comment:
Not that it does any harm, but shouldn't this `TrimEnd()` be unnecessary since it's been called on each line anyway?

Reply:
It's necessary because we `TrimEnd` the contents of each line, before re-appending it with `AppendCrlfLine`.

This means without the final trim, we end up with an additional line break right at the end of the string, which isn't necessary. If we remove this, all the tests basically fail!